src/component/Body/Weather/index.tsx

```diff
@@ -1,46 +1,39 @@
 import React from 'react';
-import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { faSync } from '@fortawesome/free-solid-svg-icons';
-import { connect } from 'react-redux';
+import { connect, ConnectedProps } from 'react-redux';
 
-import WeatherService from '../../../service/weather';
 import DateTimer from '../../../util/dateTime';
-import { getWeather } from '../../../redux/actions/index';
+import { GetWeather } from '../../../redux/actions/index';
 import './Weather.css';
+import { Props, State } from '../../../types';
+import { Icon, Temp, Card, Title, RefreshBtn } from './weatherComponent';
 
-function Icon(props) {
-  const iconUrl = `url('http://openweathermap.org/img/wn/${props.code}@4x.png'`;
-  return <div className='icon' style={{ backgroundImage: iconUrl }} />;
-}
-
-function Temp(props) {
-  return <div className='temp'>{props.children}°C</div>;
-}
+const mapStateToProps = (state: { data: any; loading: boolean }) => {
+  return {
+    weatherData: state.data,
+    isLoading: state.loading,
+  };
+};
+//mapDispatchToProps
+const actionCreators = {
+  getWeather: GetWeather,
+};
 
-function Card(props) {
-  return <div className='wtCard'>{props.children}</div>;
-}
+const connector = connect(mapStateToProps, actionCreators);
 
-function Title(props) {
-  return <div className='cardTitle'>{props.children}</div>;
-}
+type PropsFromRedux = ConnectedProps<typeof connector>;
+interface WeatherProps extends PropsFromRedux, Props {}
```

Comment:
Empty interface is not allowed here. If you just combine those types, use Union types.